packages/atomic/src/components/facets-v1/atomic-facet/atomic-facet.tsx

```diff
@@ -115,6 +124,17 @@ export class AtomicFacet implements InitializableComponent {
     };
   }
 
+  public componentShouldUpdate(next: unknown, prev: unknown, propName: string) {
+    if (propName === 'facetState') {
+      return shouldUpdateFacetSearchComponent(
```

Comment:
this is helpful for switching seamlessly between actual values and results, we don't want any flashing e.g. `Facet Values -> Empty Space -> Search Results` or `No Matches -> Facet Values -> No Matches`

Reply:
So this function compares the previous and next state to check whether or not they differ and the page should be rerendered?

Reply:
Yes, if the component should be re-rendered